include/quicr/moq_message_types.h

```diff
@@ -0,0 +1,289 @@
+
+
+#include <string>
+#include <vector>
+#include <quicr/uvarint.h>
+#include <quicr/quicr_common.h>
+#include <quicr/message_buffer.h>
+
+namespace quicr::messages {
+
+  using Version = uintVar_t;
+  using TrackNamespace = quicr::bytes;
+  using TrackName = quicr::bytes;
+  using ErrorCode = uintVar_t;
+  using StatusCode = uintVar_t;
+  using ReasonPhrase = quicr::bytes;
+  using GroupId = uintVar_t;
+  using ObjectId = uintVar_t;
+  using ObjectPriority = uintVar_t;
+  using SubscribeId = uintVar_t;
+  using TrackAlias = uintVar_t;
+  using ParamType = uintVar_t;
+
+  // Ref: https://moq-wg.github.io/moq-transport/draft-ietf-moq-transport.html#name-messages
+  constexpr uint8_t MESSAGE_TYPE_OBJECT_STREAM            = 0x0;
+  constexpr uint8_t MESSAGE_TYPE_OBJECT_DATAGRAM          = 0x1;
+  constexpr uint8_t MESSAGE_TYPE_SUBSCRIBE                = 0x3;
+  constexpr uint8_t MESSAGE_TYPE_SUBSCRIBE_OK             = 0x4;
+  constexpr uint8_t MESSAGE_TYPE_SUBSCRIBE_ERROR          = 0x5;
+  constexpr uint8_t MESSAGE_TYPE_ANNOUNCE                 = 0x6;
+  constexpr uint8_t MESSAGE_TYPE_ANNOUNCE_OK              = 0x7;
+  constexpr uint8_t MESSAGE_TYPE_ANNOUNCE_ERROR           = 0x8;
+  constexpr uint8_t MESSAGE_TYPE_UNANNOUNCE               = 0x9;
+  constexpr uint8_t MESSAGE_TYPE_UNSUBSCRIBE              = 0xA;
+  constexpr uint8_t MESSAGE_TYPE_SUBSCRIBE_DONE           = 0xB;
+  constexpr uint8_t MESSAGE_TYPE_ANNOUNCE_CANCEL          = 0xC;
+  constexpr uint8_t MESSAGE_TYPE_TRACK_STATUS_REQUEST     = 0xD;
+  constexpr uint8_t MESSAGE_TYPE_TRACK_STATUS             = 0xE;
+  constexpr uint8_t MESSAGE_TYPE_GOAWAY                   = 0x10;
+  constexpr uint8_t MESSAGE_TYPE_CLIENT_SETUP             = 0x40;
+  constexpr uint8_t MESSAGE_TYPE_SERVER_SETUP             = 0x41;
+  constexpr uint8_t MESSAGE_TYPE_STREAM_HEADER_TRACK      = 0x50;
+  constexpr uint8_t MESSAGE_TYPE_STREAM_HEADER_GROUP      = 0x51;
+
+  // TODO (Suhas): rename it to StreamMapping
+  enum ForwardingPreference : uint8_t {
+    StreamPerGroup = 0,
+    StreamPerObject,
+    StreamPerPriority,
+    StreamPerTrack,
+    Datagram
+  };
+
+  //
+  // Parameters
+  //
+
+  enum struct ParameterType : uint8_t {
+    Role = 0x0,
+    Path = 0x1,
+    AuthorizationInfo = 0x2, // version specific, unused
+  };
+
+  struct MoqParameter {
+    ParamType param_type;
+    uintVar_t param_length;
+    bytes param_value;
+  };
+
+  //
+  // Setup
+  //
+
+  struct MoqClientSetup {
+    std::vector<Version> supported_versions;
+    MoqParameter role_parameter;
+    MoqParameter path_parameter;
+  };
+
+  struct MoqServerSetup {
+    Version supported_version;
+    MoqParameter role_parameter;
+    MoqParameter path_parameter;
+  };
+
+  MessageBuffer& operator<<(MessageBuffer &buffer, const MoqParameter &msg);
+  MessageBuffer& operator>>(MessageBuffer &buffer, MoqParameter &msg);
+  MessageBuffer& operator<<(MessageBuffer &buffer, const MoqClientSetup &msg);
```

Comment:
None of these support byte stream input. Please update to support a byte stream for all message decodes. Transport will only provide bytes as input, which will be 1 byte or more at a time.